plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { ClusterDetails, KubernetesClustersSupplier } from '../types/types';
+import { KubernetesProxy, KubernetesProxyServices } from './KubernetesProxy';
+
+import { Request } from 'express';
+
+import 'buffer';
+
+jest.mock('node-fetch');
+const { Response } = jest.requireActual('node-fetch');
+
+import fetch from 'node-fetch';
+
+describe('KubernetesProxy', () => {
+  // @ts-ignore
+  let clientMock: any;
+  let sut: KubernetesProxy;
+
+  const buildEncodedRequest = (
+    clustersHeader: any,
+    query: string,
+    body?: any,
+  ): Request => {
+    const encodedQuery = encodeURIComponent(query);
+    const encodedClusters = Buffer.from(
+      JSON.stringify(clustersHeader),
+    ).toString('base64');
+
+    const req = {
+      params: {
+        encodedQuery,
+      },
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Kubernetes-Clusters': encodedClusters,
```

Comment:
It's a bit of an oddball protocol, hmm, not super fond of this json + base64 thing i must admit.

Could we leverage the ability to have multiple headers?

```
Kubernetes-Cluster: cluster-name-1=some-token
Kubernetes-Cluster: cluster-name-2=other-token
```

Reply:
I agree that a different protocol could make sense, this is just based on what was discussed in [the RFC](https://github.com/backstage/backstage/issues/12231) - not sure if it would make sense to amend the RFC at this point or if there's a better way to discuss different auth header options